### PR TITLE
implement get_node_health()

### PR DESCRIPTION
**Pull request by jmcook1186**
implements the `get_health` endpoint.

### CODE REVIEW

File: src/api_client.rs [hunk 1]
```diff
@@ -395,8 +396,19 @@ impl Client {
...
+    pub async fn get_health(&self) -> Result<HealthStatus, Error> {
+        let path = "eth/v1/node/health";
+        let target = self.endpoint.join(path)?;
+        let request = self.http.get(target);
+        let response = request.send().await?;
+        let status: u16 = StatusCode::as_u16(&response.status());
+        let result = match status {
+            200 => HealthStatus::Ready,
+            206 => HealthStatus::Syncing,
+            503 => HealthStatus::NotInitialized,
```
Comment (ralexstokes) [inline src/api_client.rs]: can match on the hard-coded constants defined in `http` like

```suggestion
            http::OK => HealthStatus::Ready,
```

and skip the conversion to a `u16`

File: src/types.rs [hunk 2]
```diff
@@ -278,6 +278,19 @@ pub enum HealthStatus {
...
+impl fmt::Display for HealthStatus {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let printable = match *self {
+            HealthStatus::Ready => "Ready",
+            HealthStatus::Syncing => "Syncing",
+            HealthStatus::NotInitialized => "Not Initialized",
+            HealthStatus::Unknown => "Unknown",
+        };
+        write!(f, "{}", printable)
+    }
```
Comment (ralexstokes) [inline src/types.rs]: do we need this?

**Review by ralexstokes — COMMENTED** [review]
nice job!

left some questions/feedback

File: src/types.rs (hunk 2 above)
Comment (jmcook1186) [reply src/types.rs]: no, we can `#derive[Debug]` on the enum instead. I erred between these two options and landed on this one, but can switch it.

File: src/api_client.rs (hunk 1 above)
Comment (jmcook1186) [reply src/api_client.rs]: ah ok, got it

File: src/types.rs (hunk 2 above)
Comment (jmcook1186) [reply src/types.rs]: done in latest commit

File: src/types.rs [hunk 3]
```diff
@@ -273,11 +273,12 @@ pub struct SyncStatus {
     pub is_syncing: bool,
 }
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug)]
```
Comment (ralexstokes) [inline src/types.rs:276]: do we even need the `Debug` impl?

File: src/api_client.rs [hunk 4]
```diff
@@ -395,8 +395,18 @@ impl Client {
...
-        unimplemented!("")
+    pub async fn get_health(&self) -> Result<HealthStatus, Error> {
+        let path = "eth/v1/node/health";
+        let target = self.endpoint.join(path)?;
+        let request = self.http.get(target);
+        let response = request.send().await?;
+        let result = match response.status() {
+            http::StatusCode::OK => HealthStatus::Ready,
+            http::StatusCode::PARTIAL_CONTENT => HealthStatus::Syncing,
+            http::StatusCode::SERVICE_UNAVAILABLE => HealthStatus::NotInitialized,
```
Comment (ralexstokes) [inline src/api_client.rs]: minor stylistic preference to `use http::StatusCode;` at the top of the file and then just match on `StatusCode::X`

**Review by ralexstokes — COMMENTED** [review]
looking good! not sure why we need the `Debug` impl and I left a small nitpick on the `StatusCode` thing

File: src/types.rs (hunk 3 above)
Comment (jmcook1186) [reply src/types.rs:276]: only to print the result to the console for testing.

File: src/api_client.rs (hunk 4 above)
Comment (jmcook1186) [reply src/api_client.rs]: ah yea ok. done.

File: src/types.rs (hunk 3 above)
Comment (ralexstokes) [reply src/types.rs:276]: ok fine

**Review by ralexstokes — APPROVED** [review]
lgtm!

**Pull request merged**